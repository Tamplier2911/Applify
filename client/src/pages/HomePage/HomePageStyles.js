import styled from "styled-components";

export const HomePageContainer = styled.section`
  display: grid;
  grid-template-rows: 70vh 1fr;
  // grid-row-gap: 10rem;
  grid-row-gap: 20rem;

  @media only screen and (max-width: 980px) {
    grid-template-rows: 68vh 1fr;
    grid-row-gap: 4rem;
    // grid-row-gap: 0rem;
    // grid-row-gap: 10rem;
  }

  @media only screen and (max-width: 768px) {
    grid-template-rows: 65vh 1fr;
  }

  @media only screen and (max-width: 720px) {
    grid-template-rows: 60vh 1fr;
    grid-row-gap: 7rem;
  }

  @media only screen and (max-width: 540px) {
    margin-top: 2rem;
    grid-template-rows: 80vh 1fr;
  }
`;

export const HomePagePlaceholder = styled.div``;
