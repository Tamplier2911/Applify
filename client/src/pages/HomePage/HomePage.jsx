// import "./HomePage.scss";
import React from "react";

// components
import Hero from "../../components/Hero/Hero";
import HomeCards from "../../components/HomeCards/HomeCards";

// JS Rendering CSS
import { HomePageContainer, HomePagePlaceholder } from "./HomePageStyles";

const HomePage = () => {
  return (
    <HomePageContainer>
      <Hero />
      <HomeCards />
      <HomePagePlaceholder />
    </HomePageContainer>
  );
};

export default HomePage;
