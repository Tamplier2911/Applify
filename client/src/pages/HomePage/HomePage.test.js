import React from "react";
import toJson from "enzyme-to-json";

// component
import HomePage from "./HomePage";

// utils
import { setShallow, findByTestAttr } from "../../tests/testUtils";

describe("<HomePage />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<HomePage />);
  });

  it("renders without error", () => {
    const homePage = findByTestAttr(wrapper, "home-page");
    expect(homePage.length).toBe(1);
  });

  it("renders childrens", () => {
    const heroComponent = findByTestAttr(wrapper, "home-page-hero");
    expect(heroComponent.length).toBe(1);

    const cardsComponent = findByTestAttr(wrapper, "home-page-cards");
    expect(cardsComponent.length).toBe(1);
  });

  it("matches snapshot", () => {
    const homePage = findByTestAttr(wrapper, "home-page");
    expect(toJson(homePage)).toMatchSnapshot();
  });
});
