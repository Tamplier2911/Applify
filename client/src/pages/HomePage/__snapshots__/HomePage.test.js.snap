// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> matches snapshot 1`] = `
<styled.section
  data-test="home-page"
>
  <Connect(Hero)
    data-test="home-page-hero"
  />
  <Connect(HomeCards)
    data-test="home-page-cards"
  />
  <styled.div />
</styled.section>
`;
